assets/js/base/context/cart-checkout/payment-methods/types.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { emitterCallback } from '../../shared/event-emit';
+import { STATUS } from './constants';
+
+type ObjectType = Record< string, unknown >;
+type EmptyObjectType = Record< string, never >;
+
+export interface PaymentMethodConfig {
+	// A unique string to identify the payment method client side.
+	name: string;
+	// A react node for your payment method UI.
+	content: ReactNode;
+	// A react node to display a preview of your payment method in the editor.
+	edit: ReactNode;
+	// A callback to determine whether the payment method should be shown in the checkout.
+	canMakePayment: (
+		cartData: ObjectType
```

Comment:
Not necessary now, but I think cartData can be defined in more detail if we look at what we send
```
{
			cartTotals,
			cartNeedsShipping,
			billingData,
			shippingAddress,
			selectedShippingMethods,
			paymentRequirements,
		}
```

Reply:
Yeah. I won't make that change now because of scope, but you're right.